src/app/external-cluster-wizard/steps/external-cluster/provider/aks/component.ts

```diff
@@ -145,39 +169,78 @@ export class AKSClusterSettingsComponent
         Validators.min(this.AUTOSCALING_MIN_VALUE),
       ]),
     });
-
     this.control(Controls.Mode).disable();
   }
 
   private _initSubscriptions(): void {
     this.form.valueChanges.pipe(takeUntil(this._unsubscribe)).subscribe(_ => {
-      this._updateExternalClusterModel();
+      this.isDialogView() ? this._updateExternalMachineDeployment() : this._updateExternalClusterModel();
       this._externalClusterService.isClusterDetailsStepValid = this.form.valid;
+      this._externalMachineDeploymentService.isAddMachineDeploymentFormValid = this.form.valid;
     });
 
-    this.control(Controls.Location)
-      .valueChanges.pipe(debounceTime(this._debounceTime))
-      .pipe(switchMap(location => this._getAKSVmSizes(location)))
-      .pipe(takeUntil(this._unsubscribe))
-      .subscribe((vmSizes: string[]) => {
-        this.vmSizes = vmSizes;
-      });
+    if (this.isDialogView()) {
+      this.control(Controls.Name).clearValidators();
+      this.control(Controls.Location).clearValidators();
+      this.control(Controls.NodeResourceGroup).clearValidators();
+      this.control(Controls.Mode).enable();
+      this._getAKSVmSizesForCreateMachineDeployment(this.cluster.spec.aksclusterSpec.location).subscribe(
+        (vmSizes: string[]) => {
+          this.vmSizes = vmSizes;
+        }
+      );
+      this._getAKSAvailableNodePoolVersionsForCreateMachineDeployment().subscribe(
+        (versions: AKSNodePoolVersionForMachineDeployments[]) => {
+          this.nodePoolVersions = versions.map(versions => versions.version);
```

Comment:
You can add type if you want but `nodePoolVersion` makes more context we are getting node pool version from versions list.

```suggestion
          this.nodePoolVersions = versions.map(nodePoolVersion => nodePoolVersion.version)
```